packages/flutter/lib/src/semantics/semantics.dart

```diff
@@ -2776,6 +2782,95 @@ class SemanticsOwner extends ChangeNotifier {
   String toString() => describeIdentity(this);
 }
 
+/// A string that carries a list of [StringAttribute].
+@immutable
+class AttributedString {
+  /// Created Attributed string.
+  ///
+  /// the [TextRange] in the [attributes] must be in side the length of the
+  /// [string].
+  const AttributedString(
+    this.string, {
+    this.attributes,
+  });
+
+  /// The plain string.
+  final String string;
+
+  /// The attributes this string carries.
+  final List<StringAttribute>? attributes;
+
+  @override
+  bool operator ==(Object other) {
+    if (other.runtimeType != runtimeType)
+      return false;
+    return other is AttributedString
+        && other.string == string
+        && other.attributes == attributes;
+  }
+
+  @override
+  int get hashCode => string.hashCode ^ attributes.hashCode;
+}
+
+/// The attribute type that is used in [StringAttribute].
+enum StringAttributeType {
+  /// The string should be spell out character by character.
+  spellOut,
+}
+
+/// The attribute that affects how assistive technologies treat the text.
+class StringAttribute {
```

Comment:
Wondering if this would be easier to use if there where an abstract StringAtribute base class and concrete SpellOutStringAttribute subclasses.